src/engraving/dom/editcapo.cpp

```diff
@@ -0,0 +1,296 @@
+    int fret = note->fret();
+    if (CapoParams::TransposeMode::PLAYBACK_ONLY != ctx.params.transposeMode) {
+        if (muse::contains(ctx.params.ignoredStrings, (string_idx_t)note->string())) {
+            // Already handled
+            if (pitch == ctx.stringData->getPitch(string, fret, 0)) {
+                return;
+            }
+            if (CapoParams::TransposeMode::STANDARD_ONLY == ctx.params.transposeMode) {
+                note->setPitch(ctx.stringData->getPitch(note->string(), note->fret(), ctx.staff));
+            } else {
+                // The safest way
+                // We're in the tab-only mode and fret now is real fret - capo fret.
+                // If string is ignored, reverse that.
+                note->setFret(fret + ctx.params.fretPosition);
+            }
+            note->setTpcFromPitch();
+            return;
+        }
+        if (pitch != ctx.stringData->getPitch(string, fret, -ctx.params.fretPosition)) {
+            if (CapoParams::TransposeMode::STANDARD_ONLY == ctx.params.transposeMode) {
+                note->setPitch(ctx.stringData->getPitch(note->string(), note->fret(), -ctx.params.fretPosition));
+            } else {
+                ctx.stringData->convertPitch(pitch, -ctx.params.fretPosition, &string, &fret);
+                note->setString(string);
+                note->setFret(fret);
+            }
+        }
+    }
+    note->setTpcFromPitch();
+}
+
+void
+EditCapo::handleModeChange(const CapoParams& oldParams, const CapoParams& newParams, int startTick, int endTick,
+                           UpdateCtx& ctx)
+{
+    switch (oldParams.transposeMode) {
+    case CapoParams::TransposeMode::PLAYBACK_ONLY:
+        if (newParams.transposeMode == CapoParams::TransposeMode::STANDARD_ONLY) {
+            ctx.noteOffset = newParams.fretPosition;
+        } else if (newParams.transposeMode == CapoParams::TransposeMode::TAB_ONLY) {
+            ctx.fretOffset = -newParams.fretPosition;
+            ctx.updateFrets = true;
+        }
+        break;
+    case CapoParams::TransposeMode::STANDARD_ONLY:
+        if (newParams.transposeMode == CapoParams::TransposeMode::PLAYBACK_ONLY) {
+            ctx.noteOffset = -oldParams.fretPosition;
+        } else if (newParams.transposeMode == CapoParams::TransposeMode::TAB_ONLY) {
+            ctx.noteOffset = -oldParams.fretPosition;
+            ctx.fretOffset = -newParams.fretPosition;
+            ctx.updateFrets = true;
+        }
+        break;
+    case CapoParams::TransposeMode::TAB_ONLY:
+        if (newParams.transposeMode == CapoParams::TransposeMode::PLAYBACK_ONLY) {
+            ctx.updateFrets = true;
+        } else if (newParams.transposeMode == CapoParams::TransposeMode::STANDARD_ONLY) {
+            ctx.noteOffset = newParams.fretPosition;
+            ctx.fretOffset = -newParams.fretPosition;
+            ctx.updateFrets = true;
+        }
+        break;
+    }
+    EditCapo::applyCapoTranspose(startTick, endTick, ctx);
```

Comment:
You're already in the EditCapo class so you don't need to write EditCapo:: (same in a few other places below)